selectors/timetableSelectors.js

```diff
@@ -0,0 +1,56 @@
+import _ from "lodash"
+import { createSelector } from 'reselect'
+
+import { LocalisationManager } from "../lib"
+
+export const timetableSelector = (state) => state.timetable.weeklyTimetable
+
+const padTimetable = (timetable) => {
+  const differences = timetable.map(([{ dateISO }], i) => {
+    if (i === 0) {
+      return 0
+    }
+    return (
+      LocalisationManager.parseToMoment(dateISO)
+        .diff(
+          LocalisationManager.parseToMoment(timetable[i - 1][0].dateISO),
+          `weeks`,
+        )
+    )
+  })
+
+  return timetable.reduce(
+    (arr, v, i) => {
+      if (differences[i] > 1) {
+        return arr
+          .concat([...new Array(differences[i] - 1)].map(() => null))
+          .concat([v])
+      }
+      return arr.concat([v])
+    },
+    [],
+  )
+}
+
+export const weeklyTimetableArraySelector = createSelector(
+  [timetableSelector],
+  (timetable) => [
+    null,
+    ...padTimetable(
+      _.chunk(
+        Object.entries(timetable)
```

Comment:
I get a "Cannot convert undefined or null to object" error when I try to run this locally

Reply:
Added 
```
(timetable) => {
    if(typeof timetable==="undefined") {
      return []
    } else {
``` 
and it seems to work fine looks awesome